src/input/manager.rs

```diff
@@ -1601,150 +1317,129 @@ impl Manager {
-        let (watcher_tx, mut watcher_rx) = mpsc::channel(BUFFER_SIZE);
-        if std::path::Path::new(DEV_PATH).exists() {
-            let tx = watcher_tx.clone();
-            tokio::task::spawn_blocking(move || {
-                log::info!("Started hidraw device discovery thread");
-                watcher::watch(DEV_PATH.into(), tx)
-            });
-        }
-        if std::path::Path::new(INPUT_PATH).exists() {
-            let tx = watcher_tx.clone();
-            tokio::task::spawn_blocking(move || {
-                log::info!("Started evdev device discovery thread");
-                watcher::watch(INPUT_PATH.into(), tx)
-            });
-        }
+        })
+    }
 
-        task::spawn(async move {
-            'outer: while let Some(event) = watcher_rx.recv().await {
-                match event {
-                    WatchEvent::Create { name, base_path } => {
-                        let subsystem = {
-                            match base_path.as_str() {
-                                "/dev" => {
-                                    if !name.starts_with("hidraw") {
-                                        None
-                                    } else {
-                                        Some("hidraw")
-                                    }
+    async fn watch_hidraw_devices(
+        cmd_tx: mpsc::Sender<ManagerCommand>,
+        watcher_rx: &mut mpsc::Receiver<WatchEvent>,
+    ) -> () {
+        'outer: while let Some(event) = watcher_rx.recv().await {
+            match event {
+                WatchEvent::Create { name, base_path } => {
+                    let subsystem = {
+                        match base_path.as_str() {
+                            "/dev" => {
+                                if !name.starts_with("hidraw") {
+                                    None
+                                } else {
+                                    Some("hidraw")
                                 }
-                                "/dev/input" => Some("input"),
-
-                                _ => None,
-                            }
-                        };
-                        let Some(subsystem) = subsystem else {
-                            log::trace!("No supported subsystem detected for {base_path}/{name}");
-                            continue;
-                        };
-
-                        // Wait until the device has initialized with udev
-                        const MAX_TRIES: u8 = 80;
-                        let mut attempt: u8 = 0;
-                        loop {
-                            // Break after max attempts reached
-                            if attempt > MAX_TRIES {
-                                log::warn!("Unable to create initialized UdevDevice for {base_path}/{name} after {MAX_TRIES} attempts.");
-                                continue 'outer;
                             }
+                            "/dev/input" => Some("input"),
```

Comment:
This function watches both hidraw(/dev/hidraw#) and evdev (/dev/input/event#), so the function name is misleading. Maybe call it watch_devnode_devices, watch_inofity_devices, something similar?
@ShadowApex opinion?

Reply:
I have named it "watch_devnodes" since it watches /dev and associate new devnodes with the corresponding udev device.